### PR TITLE
Add some winona support

### PR DESCRIPTION
**Pull request by HubertZhang**
- Add cooldown info for winona's hats
- Show recharging info when fueled's rate is negative

### CODE REVIEW

File: scripts/descriptors/fueled.lua
Comment (penguin0616) [inline scripts/descriptors/fueled.lua:1]: fuel_time is a nil value here

File: scripts/language/english.lua [hunk 1]
```diff
@@ -1400,4 +1400,16 @@ return {
 
 	-- yotb_sewer.lua
 	yotb_sewer = "Will finish sewing in: %s",
-}
\ No newline at end of file
+
+	-- inspectacleshat.lua
+	inspectacleshat = {
+		ready_to_use = "Ready to inspect",
+		cooldown = "Cooldown: %s",
```
Comment (penguin0616) [inline scripts/language/english.lua]: The cooldown in all the **inspectableshat** (for all languages) can be removed to reference the generic cooldown

File: scripts/prefab_descriptors/inspectacleshat.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		return nil
+	end
+
+	if not player.components.inspectaclesparticipant.cooldowntask then
+		return {
+			priority = 0,
+			name = "inspectacleshat",
+			description = context.lstr.inspectacleshat.ready_to_use,
+		}
+	else
```
Comment (penguin0616) [inline scripts/prefab_descriptors/inspectacleshat.lua:37]: This entire return logic here can and should be condensed. No reason to separate on a condition like that.

File: scripts/prefab_descriptors/roseglasseshat.lua [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+		return nil
+	end
+
+	if not player.components.roseinspectableuser.cooldowntask then
+		return {
+			priority = 0,
+			name = "roseglasseshat",
+			description = context.lstr.roseglasseshat.ready_to_use,
+		}
+	else
```
Comment (penguin0616) [inline scripts/prefab_descriptors/roseglasseshat.lua:36]: Same thing as the inspectacleshat.

**Review by penguin0616 — CHANGES_REQUESTED** [review]

File: scripts/descriptors/fueled.lua
Comment (HubertZhang) [reply scripts/descriptors/fueled.lua:1]: Fixed

**Review by penguin0616 — APPROVED** [review]

File: scripts/prefab_descriptors/roseglasseshat.lua [hunk 4]
```diff
@@ -0,0 +1,26 @@
+local function Describe(self, context)
```
Comment (penguin0616) [inline scripts/prefab_descriptors/roseglasseshat.lua:1]: Last thing is for you to cede this file by putting the Insight copyright at the top - you can copy it from pretty much any other file that has it.

File: scripts/prefab_descriptors/inspectacleshat.lua [hunk 5]
```diff
@@ -0,0 +1,27 @@
+local function Describe(self, context)
```
Comment (penguin0616) [inline scripts/prefab_descriptors/inspectacleshat.lua:22]: Last thing is for you to cede this file by putting the Insight copyright at the top - you can copy it from pretty much any other file that has it.

File: scripts/prefab_descriptors/roseglasseshat.lua (hunk 4 above)
Comment (penguin0616) [inline scripts/prefab_descriptors/roseglasseshat.lua:22]: Last thing is for you to cede this file by putting the Insight copyright at the top - you can copy it from pretty much any other file that has it.

**Review by penguin0616 — CHANGES_REQUESTED** [review]

**Comment by penguin0616** [comment]
Thanks for making the requested changes!
I've made one last change request, after which this PR should be good to go.

File: scripts/prefab_descriptors/roseglasseshat.lua (hunk 4 above)
Comment (HubertZhang) [reply scripts/prefab_descriptors/roseglasseshat.lua:22]: Done

**Pull request merged**